src/toil/jobStores/aws/jobStore.py

```diff
@@ -744,11 +744,25 @@ def bucket_creation_pending(error):
                         bucketExisted = False
                         logger.debug("Bucket '%s' does not exist.", bucket_name)
                         if create:
-                            logger.debug("Creating bucket '%s'.", bucket_name)
-                            location = region_to_bucket_location(self.region)
-                            bucket = self.s3_resource.create_bucket(
-                                Bucket=bucket_name,
-                                CreateBucketConfiguration={'LocationConstraint': location})
+                            location = self.region
+                            logger.debug(
+                                "Creating bucket '%s' in region %s.",
+                                bucket_name,
+                                location,
+                            )
+                            if (
+                                location == "us-east-1"
+                            ):  # see https://github.com/boto/boto3/issues/125
+                                bucket = self.s3_resource.create_bucket(
+                                    Bucket=bucket_name
+                                )
+                            else:
+                                bucket = self.s3_resource.create_bucket(
+                                    Bucket=bucket_name,
+                                    CreateBucketConfiguration={
+                                        "LocationConstraint": location
+                                    },
+                                )
```

Comment:
Since @w-gao is also going to be creating buckets, should this be pulled out into a function (`toil.lib.aws.utils.create_s3_bucket()`)?

Then we could add a unit test for that and avoid running all the AWS job store tests again in a second region.